tests/rptest/services/kcat_consumer.py

```diff
@@ -0,0 +1,274 @@
+            self._cmd += ["-C", "-t", f"{topic}"]
+        else:
+            assert partition is None, "Partition argument is not used in high-level consumer mode"
+            self._cmd += ["-G", cgroup_name]
+            trailopts += [topic]
+        if auto_commit_interval_ms is None:
+            self._cmd += ["-X", "enable.auto.commit=false"]
+        else:
+            self._cmd += [
+                "-X", "enable.auto.commit=true", "-X",
+                f"auto.commit.interval.ms={auto_commit_interval_ms}"
+            ]
+        if partition:
+            self._cmd += ["-p", f"{partition}"]
+        assert (offset is
+                None) or (first_timestamp is
+                          None), "Cannot specify both offset and timestamp"
+        if offset is not None:
+            # <value>  (absolute offset)
+            if isinstance(offset, int):
+                self._cmd += ["-o", f"{offset}"]
+            elif isinstance(offset, KcatConsumer.OffsetMeta):
+                self._cmd += ["-o", offset.value]
+            else:
+                assert False, "Offset must be an integer or an OffsetMeta"
+        if first_timestamp is not None:
+            # s@<value> (timestamp in ms to start at)
+            self._cmd += ["-o", f"s@{first_timestamp}"]
+        if num_msgs is not None:
+            self._cmd += ["-c", f"{num_msgs}"]
+        if getattr(self._redpanda, "sasl_enabled", lambda: False)():
+            cfg = self._redpanda.security_config()
+            cmd += [
+                "-X", f"security.protocol={cfg['security_protocol']}", "-X"
+                f"sasl.mechanism={cfg['sasl_mechanism']}", "-X",
+                f"sasl.username={cfg['sasl_plain_username']}", "-X",
+                f"sasl.password={cfg['sasl_plain_password']}"
+            ]
+            if cfg['sasl_mechanism'] == "GSSAPI":
+                cmd += [
+                    "-X", "sasl.kerberos.service.name=redpanda",
+                    '-Xsasl.kerberos.kinit.cmd=kinit client -t /var/lib/redpanda/client.keytab'
+                ]
+        self._cmd += trailopts
+
+    class SSHCapturedPipes:
+        def __init__(self, stdin: paramiko.channel.ChannelStdinFile,
+                     stdout: paramiko.channel.ChannelFile,
+                     stderr: paramiko.channel.ChannelStderrFile, cmd: str,
+                     logger) -> None:
+            self.stdin = stdin
+            self.stdout = stdout
+            self.stderr = stderr
+            self._cmd = cmd
+            self.logger = logger
+
+        def __enter__(self) -> 'KcatConsumer.SSHCapturedPipes':
+            return self
+
+        def __exit__(self, exc_type, exc_value, traceback) -> None:
+            try:
+                exit_status = self.stdout.channel.recv_exit_status()
+                if exit_status != 0:
+                    if self.logger is not None:
+                        #self._log(logging.DEBUG,
+                        self._redpanda.logger.debug(
```

Comment:
you're passed a logger but then you don't use it and use `self._redpanda.logger`